src/CloseButton/CloseButton.js

```diff
@@ -72,13 +73,27 @@ class CloseButton extends PureComponent {
         {...generateDataAttr(this.props, ['skin', 'size'])}
         data-hook={dataHook}
       >
-        {children
-          ? React.cloneElement(children, {
-              size: childSize,
-              width: childSize,
-              height: childSize,
-            })
-          : this._renderCloseIcon(size)}
+        {children ? (
+          React.cloneElement(children, {
+            size: childSize,
+            width: childSize,
+            height: childSize,
+          })
+        ) : (
+          <ThemeProviderConsumerBackwardCompatible
+            defaultIcons={{
+              CloseButton: {
+                small: Close,
+                medium: CloseLarge,
+                large: CloseLarge,
+              },
+            }}
+          >
```

Comment:
```suggestion
          <ThemeContext.Consumer>
```

i think this is how ideally it should look like to use the theme context. Default values should be set in `React.createContext({ icons: { CloseButton: ... } })`

Reply:
If I understand you correctly, this means to put all icons in a single file, and then you simply use it from the context.

I tried that, but the problem is that it will immediately increase the bundle size of each component since you will bundle all the icons, even if you're not using them.

You can see [this commit](https://github.com/wix/wix-style-react/pull/5976/commits/965c6810196545479d4fee0ed4ce3a5970dd7436). I created a `defaultIcons` file and used it [here in the ThemeProviderConsumerBackwardCompatible](https://github.com/wix/wix-style-react/pull/5976/commits/965c6810196545479d4fee0ed4ce3a5970dd7436#diff-02e304bd6251cb0ff409ad88bc44e39aR13)
But then I had to ditch it and define the default icon per component.